homeworks/franktoth/5_oop/car_fleet_mgr.py

```diff
@@ -0,0 +1,71 @@
+#Creating Car class with the given attributes and methods
+
+class Car:
+    def __init__(self, brand, model, year):
+        self.brand = brand
+        self.model = model
+        self.year = year
+        self.mileage = 0
+        self.fuel_level = 100
+
+    def drive(self, distance):
+        fuel_consumption = distance * 0.1
+        if fuel_consumption <= self.fuel_level:
+            self.mileage += distance
+            self.fuel_level -= fuel_consumption
+            print(f"Car {self.brand} {self.model} drove {distance} km.")
+        else:
+            print(f"Car {self.brand} {self.model} doesn't have enough fuel to drive {distance} km.")
+
+    def refuel(self, amount):
+        self.fuel_level = min(self.fuel_level + amount, 100)
+        print(f"Car {self.brand} {self.model} refueled. Current fuel level: {self.fuel_level}%")
+
+#Creating Fleet class with the given methods
+
+class Fleet:
+    def __init__(self):
+        self.cars = []
+
+    def add_car(self, car):
+        self.cars.append(car)
+
+    def remove_car(self, car):
```

Comment:
Ha ez a car object nincs a listában, ValueError-t kapunk. Vizsgál meg törlés előtt hogy benne van e a listában, hogy ezt elkerüld.

Reply:
Ezzel elszórakoztam egy adagot. Először is átváltottam brand és model-re, mint "elvárt" attributomokra. Közelebb van a valósághoz, mint egy object identifier. És ezek alapján csekkolja, hogy van-e ilyen auto a fleet-ben.